dali/pipeline/executor/executor.cc

```diff
@@ -461,6 +527,34 @@ int Executor<WorkspacePolicy, QueuePolicy>::InferBatchSize(
   return batch_size;
 }
 
+
+template<typename WorkspacePolicy, typename QueuePolicy>
+void Executor<WorkspacePolicy, QueuePolicy>::InitIterationData() {
+  cpu_iteration_id_ = 0;
+  mixed_iteration_id_ = 0;
+  gpu_iteration_id_ = 0;
+  output_iteration_id_ = 0;
+  size_t iteration_data_size = CalcIterationDataSize();
+  iteration_data_.resize(iteration_data_size);
+  for (auto& id : iteration_data_) {
+    id.operator_traces = std::make_shared<operator_trace_map_t>();
```

Comment:
Do we need to make it a shared_ptr?

Reply:
Yes. This is because all the workspaces in a given iteration have to know all the operator traces. I've explained it also in [this comment](https://github.com/NVIDIA/DALI/pull/4564#discussion_r1070287929). Therefore, the easiest way would be if they share the same `operator_trace_map_t`.